### DISCUSSION TITLE
Using Intel MKL Pardiso with OpenMP on Mac

### DISCUSSION THREAD

**Discussion by maddielynklein** [coin-or/Ipopt#450] [Q&A] [answered]
I am trying to set up IPOPT to use Intel MKL Pardiso using openmp for parallelization on mac OSX but cannot figure out the correct linker/compiler flags. I've tried to use the [Intel MKL link line advisor](https://software.intel.com/content/www/us/en/develop/tools/oneapi/components/onemkl/link-line-advisor.html). This is the link line they suggest:
-L${MKLROOT}/lib -Wl,-rpath,${MKLROOT}/lib -lmkl_intel_ilp64 -lmkl_intel_thread -lmkl_core -lmkl_blacs_mpich_ilp64 -liomp5 -lpthread -lm -ldl
 But when I try using that IPOPT's configure fails with "configure: error: Could not find dsyev in Lapack".

Has anyone been successful at using the parallel implementation of MKL Pardiso with IPOPT in C++? I've tried multiple combinations of lapack flags and linker flags but have been unable to get it configured correctly.

**Comment by svigerske** [comment, marked as answer]
You may want to check the config.log file for details on whey these flags didn't work to link `dsyev`.

I seem to have used something like
```
--with-lapack="-L${MKLROOT} -lmkl_intel_lp64 -lmkl_core -lmkl_intel_thread -liomp5 -ldl -lpthread -lm"
```
